src/main/java/spark/SparkInstance.java

```diff
@@ -327,11 +327,11 @@ public synchronized void init() {
      * @param exceptionClass the exception class
      * @param handler        The handler
      */
-    public synchronized void exception(Class<? extends Exception> exceptionClass, ExceptionHandler handler) {
+    public synchronized <T extends Exception> void exception(Class<T> exceptionClass, ExceptionHandler<? super T> handler) {
```

Comment:
T is already the lower bound of the type you can have here, so I think "? super T" does not provide more information for the handler nor reduce the scope of the contravariance, don't you think so ? ExceptionHandle<T> should be fine